core/src/main/java/smile/regression/treeshap/ShapTree.java

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2019 Haifeng Li
+ *
+ * Smile is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * Smile is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Smile.  If not, see <https://www.gnu.org/licenses/>.
+ ******************************************************************************/
+
+package smile.regression.treeshap;
+
+import java.util.Arrays;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import smile.base.cart.InternalNode;
+import smile.base.cart.LeafNode;
+import smile.base.cart.Node;
+import smile.base.cart.RegressionNode;
+import smile.regression.RegressionTree;
+
+/** 
+ * structure for shap value calculation. implementation reference:
+ * 
+ * <pre>
+ * https://github.com/slundberg/shap/blob/master/shap/explainers/pytree.py
+ * </pre>
+ * 
+ * @author ray
+ */
+public class ShapTree {
+	
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(ShapTree.class);
+
+  public boolean normalize = false;
+  public int[] children_left;
+  public int[] children_right;
+  public int[] children_default;
+  public int[] features;
+  public Node[] thresholds;
+  public double[] values;
+  public double[] node_sample_weight; // equals number of samples for a node divided by total number of samples
+  public int max_depth;
+
+  public ShapTree(RegressionTree tree) {
```

Comment:
It looks to me that this class can be applied to DecisionTree too. Shall we take `smile.base.cart.CART`? If so, how about moving this class to `smile.feature` package?

Reply:
Even though it is only applicable to regression tree, I still think `smile.feature` is a better home. No need to create a subpackage `treeshap`.